gui/default/syncthing/device/editDeviceModalView.html

```diff
@@ -3,13 +3,15 @@
     <form role="form" name="deviceEditor">
       <div class="form-group" ng-class="{'has-error': deviceEditor.deviceID.$invalid && deviceEditor.deviceID.$dirty}" ng-init="loadFormIntoScope(deviceEditor)">
         <label translate for="deviceID">Device ID</label>
-        <input ng-if="!editingExisting" name="deviceID" id="deviceID" class="form-control text-monospace" type="text" ng-model="currentDevice.deviceID" required valid-deviceid list="discovery-list" />
-        <datalist id="discovery-list" ng-if="!editingExisting">
-          <option ng-repeat="(id, data) in discovery" value="{{id}}" />
-        </datalist>
+        <select ng-if="!editingExisting && !isEmptyObject(discovery)" class="form-control" ng-model="currentDevice.deviceID">
+          <option value='' translate>Enter manually below</option>
+          <option ng-repeat="(id, data) in discovery" value="{{id}}">{{id}}</option>
+        </select>
+        <input ng-if="!editingExisting && (!currentDevice.deviceID || isEmptyObject(discovery))" name="deviceID" id="deviceID" class="form-control text-monospace" type="text" ng-model="manualDeviceID" required valid-deviceid />
         <div ng-if="editingExisting" class="well well-sm text-monospace" select-on-click>{{currentDevice.deviceID}}</div>
         <p class="help-block">
-          <span translate ng-if="deviceEditor.deviceID.$valid || deviceEditor.deviceID.$pristine">The device ID to enter here can be found in the "Actions > Show ID" dialog on the other device. Spaces and dashes are optional (ignored).</span>
+          <span translate ng-if="!currentDevice.deviceID && !isEmptyObject(discovery) && deviceEditor.deviceID.$pristine">Choose a discovered device ID or enter one manually.</span>
+          <span translate ng-if="!currentDevice.deviceID && deviceEditor.deviceID.$pristine">The device ID can be found in the "Actions > Show ID" dialog on the other device. Spaces and dashes are optional (ignored).</span>
```

Comment:
I don't see the point of telling the user where to find the device ID when a valid one has already been entered or selected -> removal of $valid and checking whether an ID has been selected in from the list.